crates/bitwarden-ssh/src/error.rs

```diff
@@ -0,0 +1,11 @@
+use bitwarden_error::prelude::*;
+use thiserror::Error;
+
+#[bitwarden_error(flat)]
+#[derive(Error, Debug)]
+pub enum KeyGenerationError {
+    #[error("Failed to generate key: {0}")]
+    KeyGenerationError(String),
```

Comment:
You should be able to change these from `String` to `ssh_key::Error`, which would also simplify your `map_err` to:
```
let rsa_keypair = ssh_key::private::RsaKeypair::random(&mut rng, bits)
                .map_err(KeyGenerationError::KeyGenerationError)?;
```

Reply:
Nice, should both be fixed now I think